fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/IndexEntry.java

```diff
@@ -24,8 +24,10 @@
 import com.apple.foundationdb.record.metadata.Index;
 import com.apple.foundationdb.record.metadata.Key;
 import com.apple.foundationdb.record.metadata.Key.Evaluated.NullStandin;
+import com.apple.foundationdb.record.provider.foundationdb.FDBStoredRecord;
```

Comment:
This feels like it's messing with the separation of concerns, making what is essentially a neutral data structure know about the details of physical storage in another module.